install_config/install/deploy_router.adoc

```diff
@@ -265,6 +265,217 @@ $ iptables -A INPUT -p tcp --dport 10443 -j ACCEPT
+Suppose there are 26 routes, named `a` -- `z`,
+in the pool, with various labels:
+
+.Possible labels on routes in the pool
+----
+sla=high       geo=east     hw=modest     dept=finance
+sla=medium     geo=west     hw=strong     dept=dev
+sla=low                                   dept=ops
+----
+
+These labels express the concepts:
+service level agreement, geographical location,
+hardware requirements, and department.
+The routes in the pool can have at most one label from each column.
+Some routes may have other labels, entirely, or none at all.
+
+[options="header",cols="1,1,1,1,1,3"]
+|===
+|Name(s) |SLA |Geo |HW |Dept |Other Labels
+
+|`a`
+|`high`
+|`east`
+|`modest`
+|`finance`
+|`type=static`
+
+|`b`
+|
+|`west`
+|`strong`
+|
+|`type=dynamic`
+
+|`c`, `d`, `e`
+|`low`
+|
+|`modest`
+|
+|`type=static`
+
+|`g` -- `k`
+|`medium`
+|
+|`strong`
+|`dev`
+|
+
+|`l` -- `s`
+|`high`
+|
+|`modest`
+|`ops`
+|
+
+|`t` -- `z`
+|
+|`west`
+|
+|
+|`type=dynamic`
+
+|===
+
+Here is a convenience script *_mkshard_*  that
+ilustrates how `oadm router`, `oc env` and `oc scale`
+work together to make a router shard.
+
+====
+[source,bash]
+----
+#!/bin/bash
+# Usage: mkshard ID SELECTION-EXPRESSION
+id=$1
+sel="$2"
+router=router-shard-$id           //<1>
+oadm router $router --replicas=0  //<2>
+dc=dc/router-shard-$id            //<3>
+oc env   $dc ROUTE_LABELS="$sel"  //<4>
+oc scale $dc --replicas=3         //<5>
+exit $?
+----
+<1> The created router has name `router-shard-<id>`.
+<2> Specify no scaling for now.
+<3> The deployment configuration for the router.
+<4> Set the selection expression using `oc env`.
+    The selection expression is the value of
+    the `ROUTE_LABELS` environment variable.
+<5> Scale it up.
+====
+
+Running *_mkshard_* several times creates several routers:
+
+[options="header",cols="2,3,2"]
+|===
+|Router |Selection Expression |Routes
+
+|`router-shard-1`
+|`sla=high`
+|`a`, `l` -- `s`
+
+|`router-shard-2`
+|`geo=west`
+|`b`, `t` -- `z`
+
+|`router-shard-3`
+|`dept=dev`
+|`g` -- `k`
+
+|===
+
+
+[[modifying-router-shards]]
+=== Modifying Router Shards
+
+Because a router shard is a construct
+link:../../architecture/core_concepts/routes.html#router-sharding[based on labels],
+you can modify either the labels (via
+link:../../cli_reference/basic_cli_operations.html#application-modification-cli-operations[`oc label`])
+or the selection expression.
+
+This section extends the example started in the
+link:#creating-router-shards[Creating Router Shards] section,
+demonstrating how to change the selection expression.
+
+Here is a convenience script *_modshard_* that modifies
+an existing router to use a new selection expression:
+
+====
+[source,bash]
+----
+#!/bin/bash
+# Usage: modshard ID SELECTION-EXPRESSION...
+id=$1
+shift
+router=router-shard-$id       //<1>
+dc=dc/$router                 //<2>
+oc scale $dc --replicas=0     //<3>
+oc env   $dc "$@"             //<4>
+oc scale $dc --replicas=3     //<5>
```

Comment:
@ramr Thanks for the feedback.

I think dropping the `exit $?` is fine.  I will change that shortly.

However, i'm not convinced abut avoiding `oc scale` altogether due to the `Rolling` strategy handling that automagically.  As `Rolling` is the default deployment strategy, i'd prefer to mention in the docs that the script can be simplified by omitting those lines.  This way readers who use `Rolling` can benefit, as well as those who don't.